extras/indexing/src/main/java/mvm/rya/indexing/mongodb/AbstractMongoIndexer.java

```diff
@@ -64,23 +64,15 @@
 
     protected T storageStrategy;
 
-    /**
-     * Creates a new {@link AbstractMongoIndexer} with the provided mongo client.
-     * @param mongoClient The {@link MongoClient} to use with this indexer.
-     */
-    public AbstractMongoIndexer(final MongoClient mongoClient) {
-        this.mongoClient = checkNotNull(mongoClient);
-    }
-
     protected void init() throws NumberFormatException, IOException{
         dbName = conf.get(MongoDBRdfConfiguration.MONGO_DB_NAME);
         db = this.mongoClient.getDB(dbName);
         collection = db.getCollection(conf.get(MongoDBRdfConfiguration.MONGO_COLLECTION_PREFIX, "rya") + getCollectionName());
     }
 
-    @Override
-    public void setConf(final Configuration conf) {
-        this.conf = conf;
+    public void initIndexer(final Configuration conf, final MongoClient client) throws NumberFormatException, IOException {
+        this.mongoClient = client;
```

Comment:
The original checked for client=null.  This version will fail in init().  Maybe that is fail-fast enough.